### PR TITLE
Add support for more detailed connection metrics

### PR DESCRIPTION
**Pull request by matriv**
Expose new metrics implemented with:
https://github.com/crate/crate/pull/15866

### CODE REVIEW

File: src/test/java/io/crate/jmx/integrationtests/MetricsITest.java [hunk 1]
```diff
@@ -68,9 +68,22 @@ public void testQueryStatsMetrics() {
     public void testConnectionsMetrics() {
         assertMetricValue("crate_connections{protocol=\"psql\",property=\"open\",} ");
         assertMetricValue("crate_connections{protocol=\"psql\",property=\"total\",} ");
+        assertMetricValue("crate_connections{protocol=\"psql\",property=\"messagesreceived\",} ", -1);
```
Comment (matriv) [inline src/test/java/io/crate/jmx/integrationtests/MetricsITest.java:71]: In order to create psql connections we need psql JDBC driver, and my argument against this is that we may need to handle different versions to test against different CrateDB versions. If you think it's important to get actual values > -1 here, I can give it a try.

File: src/test/java/io/crate/jmx/integrationtests/V480MetricsITest.java [hunk 2]
```diff
@@ -30,4 +32,14 @@ public class V480MetricsITest extends MetricsITest {
     String getCrateDistributionURL() {
         return URL;
     }
+
+    @Override
+    @Test
+    public void testConnectionsMetrics() {
```
Comment (matriv) [inline src/test/java/io/crate/jmx/integrationtests/V480MetricsITest.java:38]: There is no way to get the cluster version from JMX or ` testCluster`, so to make the older versions run, need to override the test method to only check for the old stats.

File: src/test/java/io/crate/jmx/integrationtests/MetricsITest.java (hunk 1 above)
Comment (seut) [reply src/test/java/io/crate/jmx/integrationtests/MetricsITest.java:71]: We do not really need to test what values are returned here. We just want to test that the metric exists and contains a numeric value. I suggest to just adjust the `assertMetricValue` to compare against `-1` by default, I see no need to make this configurable.

File: src/test/java/io/crate/jmx/integrationtests/V550MetricsITest.java [hunk 3]
```diff
@@ -22,12 +22,24 @@
 
 package io.crate.jmx.integrationtests;
 
-public class V420MetricsITest extends MetricsITest {
+import org.junit.Test;
 
-    private static final String URL = "https://cdn.crate.io/downloads/releases/crate-4.2.0.tar.gz";
+public class V550MetricsITest extends MetricsITest {
```
Comment (seut) [inline src/test/java/io/crate/jmx/integrationtests/V550MetricsITest.java:27]: Any difference to `5.4`? Seems not to, so why do we need this test case?

File: src/test/java/io/crate/jmx/integrationtests/V560MetricsITest.java [hunk 4]
```diff
@@ -22,12 +22,24 @@
 
 package io.crate.jmx.integrationtests;
 
-public class V430MetricsITest extends MetricsITest {
+import org.junit.Test;
 
-    private static final String URL = "https://cdn.crate.io/downloads/releases/crate-4.3.0.tar.gz";
+public class V560MetricsITest extends MetricsITest {
```
Comment (seut) [inline src/test/java/io/crate/jmx/integrationtests/V560MetricsITest.java:27]: Again nothing changed for `5.4`, `5.5`, `5.6`. I'd remove...

File: src/test/java/io/crate/jmx/integrationtests/V570MetricsITest.java [hunk 5]
```diff
@@ -22,12 +22,24 @@
 
 package io.crate.jmx.integrationtests;
 
-public class V440MetricsITest extends MetricsITest {
+import org.junit.Test;
 
-    private static final String URL = "https://cdn.crate.io/downloads/releases/crate-4.4.0.tar.gz";
+public class V570MetricsITest extends MetricsITest {
```
Comment (seut) [inline src/test/java/io/crate/jmx/integrationtests/V570MetricsITest.java:27]: Same here, either extend from the others or remove. Duplication for no reason is not so great.

File: src/test/java/io/crate/jmx/integrationtests/MetricsITest.java (hunk 1 above)
Comment (matriv) [reply src/test/java/io/crate/jmx/integrationtests/MetricsITest.java:71]: :+1

File: src/test/java/io/crate/jmx/integrationtests/V560MetricsITest.java (hunk 4 above)
Comment (matriv) [reply src/test/java/io/crate/jmx/integrationtests/V560MetricsITest.java:27]: Ok, I will just keep `4.8` and `5.7` + the MetricItest which tests latest nightly.

Comment (matriv) [reply src/test/java/io/crate/jmx/integrationtests/V560MetricsITest.java]: I only kept 5.7, 4.8 supports the same metrics, so...

**Review by seut — APPROVED** [review]
👍

**Pull request merged**